app/models/like.rb

```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  belongs_to :posts
+
+  def update_post_likes_counter
+    posts.update(likescounter: posts.likes.count)
+  end
```

Comment:
- **[OPTIONAL]** I think it would be a good idea to use an [Active Record Callback](https://guides.rubyonrails.org/active_record_callbacks.html) called `after_save` and make this method private. See the snippet below.

- **[OPTIONAL]** We can improve this method by using `#increment!` method.

  ```
  after_save :update_post_likes_counter

  private

  def update_post_likes_counter
    post.increment!(:likescounter)
  end
  ```